parcels/tools/_helpers.py

```diff
@@ -152,3 +152,9 @@ def timedelta_to_float(dt: float | timedelta | np.timedelta64) -> float:
     if isinstance(dt, np.timedelta64):
         return float(dt / np.timedelta64(1, "s"))
     return float(dt)
+
+
+def should_calculate_next_ti(ti: int, tau: float, tdim: int) -> np.bool:
```

Comment:
Eh. The mypy error is because this function either returns a `np.bool` (because of `tau > np.finfo(float).eps`) or a normal bool (due to the `ti < tdim - 1` calculation).

Maybe let's just remove the `-> np.bool` so that it isn't type checked...